python/lambdas/copy-files-from-tdr/lambda_function.py

```diff
@@ -1,23 +1,79 @@
 import os
 import json
+import uuid
+from datetime import datetime
+
 import boto3
+import botocore
+import jsonschema
+from botocore.exceptions import ClientError
+from jsonschema import validate
+from tdr_json_schema import incoming_schema
 
 s3_client = boto3.client('s3')
 sqs_client = boto3.client('sqs')
 
-
 def lambda_handler(event, context):
     destination_bucket = os.environ["DESTINATION_BUCKET"]
     destination_queue = os.environ["DESTINATION_QUEUE"]
     for record in event['Records']:
         body = json.loads(record['body'])
         file_id = body['fileId']
         source_bucket = body['bucket']
+        assert_objects_exist(source_bucket, file_id)
+        validate_metadata(source_bucket, f"{file_id}.metadata")
         file_location = copy_object_to_s3(destination_bucket, file_id, source_bucket)
         copy_object_to_s3(destination_bucket, f"{file_id}.metadata", source_bucket)
         sqs_body = json.dumps({'id': file_id, 'location': file_location})
         sqs_client.send_message(QueueUrl=destination_queue, MessageBody=sqs_body)
 
+def assert_objects_exist(source_bucket, file_id):
+    try:
```

Comment:
Instead of repeating the trys, could we just add the two files to a List, pass them in and then for loop on them?

Reply:
I'm not sure the trys add anything here. If the object doesn't exist it'll log `f"Object {file_id} does not exist, underlying error is: {ex}"`, but if any other error happens it'll still log as an unhandled exception.

Reply:
Typically we only catch those specific exceptions that we can handle, hence a specific catch of ClientError and modifying the exception to include the missing id. If there are other errors thrown (e.g. Endpoint connection error or no credentials error), they are unrelated to existence of the objects, hence they should be bubbled up as they are, isn't it?

Reply:
Yea, that makes sense. I had forgotten that the stock error from AWS doesn't include the bucket and object when I wrote the comment.